MSAL/src/requests/MSALInteractiveRequest.m

```diff
@@ -176,77 +105,98 @@ - (void)acquireToken:(MSALCompletionBlock)completionBlock
+    config.extraQueryParameters = _parameters.extraQueryParameters;
+
+    _webviewConfig = config;
     
-    [MSALWebUI startWebUIWithURL:authorizationUrl
-                         context:_parameters
-                 completionBlock:^(NSURL *response, NSError *error)
-     {
-         s_currentRequest = nil;
-         if (error)
-         {
-             MSALTelemetryAPIEvent *event = [self getTelemetryAPIEvent];
-             [self stopTelemetryEvent:event error:error];
-             
-             completionBlock(nil, error);
-             return;
-         }
-         
-         if ([NSString msidIsStringNilOrBlank:response.absoluteString])
-         {
-             // This error case *really* shouldn't occur. If we're seeing it it's almost certainly a developer bug
-             ERROR_COMPLETION(_parameters, MSALErrorNoAuthorizationResponse, @"No authorization response received from server.");
-         }
-         
-         NSDictionary *params = [NSDictionary msidURLFormDecode:response.query];
-         CHECK_ERROR_COMPLETION(params, _parameters, MSALErrorBadAuthorizationResponse, @"Authorization response from the server code not be decoded.");
-         
-         CHECK_ERROR_COMPLETION([_state isEqualToString:params[MSID_OAUTH2_STATE]], _parameters, MSALErrorInvalidState, @"State returned from the server does not match");
-         
-         _code = params[MSID_OAUTH2_CODE];
-         if (_code)
-         {
-             [super acquireToken:completionBlock];
-             return;
-         }
-         
-         NSString *authorizationError = params[MSID_OAUTH2_ERROR];
-         if (authorizationError)
-         {
-             NSString *errorDescription = params[MSID_OAUTH2_ERROR_DESCRIPTION];
-             NSString *subError = params[MSID_OAUTH2_SUB_ERROR];
-             MSALErrorCode code = MSALErrorCodeForOAuthError(authorizationError, MSALErrorAuthorizationFailed);
-             MSALLogError(_parameters, MSALErrorDomain, code, errorDescription, authorizationError, subError, __FUNCTION__, __LINE__);
-             
-             NSError *msalError = MSALCreateError(MSALErrorDomain, code, errorDescription, authorizationError, subError, nil, nil);
-                          
-             MSALTelemetryAPIEvent *event = [self getTelemetryAPIEvent];
-             [self stopTelemetryEvent:event error:msalError];
-             
-             completionBlock(nil, msalError);
-             return;
-         }
-         
-         ERROR_COMPLETION(_parameters, MSALErrorBadAuthorizationResponse, @"No code or error in server response.");
-     }];
-
+    void (^webAuthCompletion)(MSIDWebviewResponse *, NSError *) = ^void(MSIDWebviewResponse *response, NSError *error)
+    {
+        if ([response isKindOfClass:MSIDWebAADAuthResponse.class])
+        {
+            MSIDWebAADAuthResponse *oauthResponse = (MSIDWebAADAuthResponse *)response;
+            
+            if (oauthResponse.authorizationCode)
```

Comment:
nit: check isNilOrEmptry for correctness

Reply:
I moved the check to the MSID section.